### PR TITLE
VSFP-130: Add NFR2 validation

### PR DESCRIPTION
**Pull request by fl3pp**

### CODE REVIEW

File: SA/design_concept/content/results/results_requirement_validation.typ [hunk 1]
```diff
@@ -54,3 +54,32 @@ As can be seen in @requirement_validation_table, all requirements related to fun
...
 Due to time constraints, the requirements #ac("UC")2, #ac("UC")7, #ac("UC")8 and #ac("UC")9 had to be considered as out-of-scope.
 However, during the implementation of the type inference engine, #ac("UC")7 was kept in mind so that custom data types could be added without much effort.
+
+=== Validation of NFR2 <validation-nfr2>
+
+The non-functional requirement NFR2, as described in @nfr2, states that a user
+that isn't familiar with functional programming should be able to use VisualFP
+within 1 hour.
+
+We validated this requirement by showcasing the #ac("PoC") to Samuel Bernhard, a
```
Comment (lstreckeisen) [inline SA/design_concept/content/results/results_requirement_validation.typ:64]:
```suggestion
This requirement was validated by showcasing the #ac("PoC") to Samuel Bernhard, a
```

File: SA/design_concept/content/results/results_requirement_validation.typ [hunk 2]
```diff
@@ -54,3 +54,32 @@ As can be seen in @requirement_validation_table, all requirements related to fun
...
+achieved.
+
+Still, the trial pointed out some aspects that are worth to be noted for future
+development on the project:
+
+- As he wasn't familiar with the function notation used by Haskell, he wasn't
+  able to make use of type hints (type-holes, function signatures, etc) without
+  additional explanation.
+
+- He also wasn't familiar with the `cons` and `nil` construction of lists, so we
```
Comment (lstreckeisen) [inline SA/design_concept/content/results/results_requirement_validation.typ:78]:
```suggestion
- He also wasn't familiar with the `cons` and `nil` construction of lists,
```

File: SA/design_concept/content/results/results_requirement_validation.typ [hunk 3]
```diff
@@ -54,3 +54,32 @@ As can be seen in @requirement_validation_table, all requirements related to fun
...
+  had to explain to him how it works before he could use it.
```
Comment (lstreckeisen) [inline SA/design_concept/content/results/results_requirement_validation.typ:79]:
```suggestion
  additional explanation was necessary before he could use it.
```

File: SA/design_concept/content/results/results_requirement_validation.typ [hunk 4]
```diff
@@ -54,3 +54,32 @@ As can be seen in @requirement_validation_table, all requirements related to fun
...
+
+- He would have liked to execute his created values.
+
+- After the showcase, we showed what outlooks we had thought of, and he liked
```
Comment (lstreckeisen) [inline SA/design_concept/content/results/results_requirement_validation.typ:83]:
```suggestion
- After the showcase, the envisioned outlooks were presented to him. He liked
```

File: SA/design_concept/content/results/results_requirement_validation.typ [hunk 5]
```diff
@@ -54,3 +54,32 @@ As can be seen in @requirement_validation_table, all requirements related to fun
...
+  the option of the visual and textual language (@outlook-visual-textual) very
+  much.
```
Comment (lstreckeisen) [inline SA/design_concept/content/results/results_requirement_validation.typ:85]:
```suggestion
  much.
  
These shortcomings could be fixed by adding a getting started tour to the application that introduces beginners to VisualFP, explaining the different UI components and a tutorial for value construction, but also on Haskell's type notation.
In addition, block values provided by VisualFP could feature a brief description.
```

Comment (fl3pp) [reply SA/design_concept/content/results/results_requirement_validation.typ]: I made some adjustments to the suggestion and pushed it

File: SA/design_concept/content/results/results_requirement_validation.typ [hunk 6]
```diff
@@ -54,3 +54,35 @@ As can be seen in @requirement_validation_table, all requirements related to fun
...
+- He also wasn't familiar with the `cons` and `nil` construction of lists,
+  additional explanation was necessary before he could use it.
+
+- He would have liked to execute his created values.
+
+- After the showcase, the envisioned outlooks were presented to him. He liked
+  the option of the visual and textual language (@outlook-visual-textual) very
+  much.
+
+These shortcomings could be fixed by adding a getting-started for beginners tour to VisualFP, which would explain the different UI components, provide a tutorial for value construction, and explain Haskell's function type notation.
```
Comment (lstreckeisen) [inline SA/design_concept/content/results/results_requirement_validation.typ:87]:
```suggestion
These shortcomings could be fixed by adding a getting-started tour for beginners, which would explain the different UI components, provide a tutorial for value construction, and explain Haskell's function type notation.
```

**Review by lstreckeisen — APPROVED** [review]

**Pull request merged**